### PR TITLE
Refactor DIM  app layer

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Changelist 

Refactored the DIM app layer to comply with the code architecture.

1. Refactored `app_avgPower.h/c` and `app_driveState.h/c` module
2. Deleted DIM world in favor of `app_globals.h/c`

### Testing Done

1. Test IMD LED: Fault IMD, make sure IMD LED is set under IMD fault. PASSED
2. Test SHDN LED: Check SHDN LED is on when contactors open, and off when contactors closed. PASSED
3. Test DRIVE LED: Put car into drive state, make sure drive LED is set. PASSED
4. Test start switch: Can put into drive state and drive. PASSED
5. Test aux switch: Read CAN msg and confirm it mirrors switch position. PASSED
6. RGB LED for BMS, DCM, FSM, DIM. PASSED 

- [x] CI passes
- [x] Tested on car

### CODE REVIEW

File: firmware/shared/src/dev/dev_rgbLed.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+{
+    hw_gpio_writePin(&rgb_led->red_gpio, red_on);
+    hw_gpio_writePin(&rgb_led->green_gpio, green_on);
+    hw_gpio_writePin(&rgb_led->blue_gpio, blue_on);
+}
+
+void dev_rgbLed_disable(const RgbLed *rgb_led)
+{
+    dev_rgbLed_enable(rgb_led, false, false, false);
+}
```
Comment (will-chaba) [inline firmware/shared/src/dev/dev_rgbLed.c:13]: nit: missing newline :laughing:

File: firmware/shared/src/io/Io_SharedSoftwareWatchdog.h
Comment (will-chaba) [inline firmware/shared/src/io/Io_SharedSoftwareWatchdog.h:1]: are we re-naming this file as well?

File: firmware/thruna/DIM/src/app/configs/App_SharedStateMachineConfig.h
Comment (will-chaba) [inline firmware/thruna/DIM/src/app/configs/App_SharedStateMachineConfig.h:1]: Assuming this will be renamed once all boards are done as well?

**Review by will-chaba — APPROVED** [review]
Looks good to me!

File: firmware/shared/src/io/Io_SharedSoftwareWatchdog.h
Comment (gtaharaedmonds) [reply firmware/shared/src/io/Io_SharedSoftwareWatchdog.h:1]: Oh yeah I'm leaving all shared stuff to a separate PR to limit scope to only the DIM (as much as possible)

File: firmware/thruna/DIM/src/app/configs/App_SharedStateMachineConfig.h
Comment (gtaharaedmonds) [reply firmware/thruna/DIM/src/app/configs/App_SharedStateMachineConfig.h:1]: Yep correct

File: firmware/shared/src/dev/dev_rgbLed.c (hunk 1 above)
Comment (gtaharaedmonds) [reply firmware/shared/src/dev/dev_rgbLed.c]: 😎

**Pull request merged**